mne_bids/mne_bids.py

```diff
@@ -528,22 +528,21 @@ def _sidecar_json(raw, task, manufacturer, fname, kind, eeg_ref=None,
     return fname
 
 
-def raw_to_bids(subject_id, task, raw_file, output_path, session_id=None,
-                acquisition=None, run=None, kind='meg', events_data=None,
-                event_id=None, hpi=None, electrode=None, hsp=None,
-                eeg_ref=None, eeg_gnd=None, config=None,
-                overwrite=False, verbose=True):
+def convert_to_bids(raw_fname, output_path, subject_id, task, session_id=None,
+                    acquisition=None, run=None, processing=None, kind='auto',
+                    events_data=None, event_id=None, overwrite=False,
+                    verbose=True, **kwargs):
```

Comment:
@agramfort @teonbrooks take a look at this new API

Just some thoughts that pop up in my head as I think about this:

- while we do not expose the quirks of the MNE I/O API anymore, the user still will have to use a dozen or so arguments when _calling_ this function. It's the cost you pay for having everything in one function.

- do we eventually also implement a `write_raw_bids` function or would you consider that off the table? 
    - Pros
         - Useful if the user wants to edit/correct some information before writing the file
         - We're not using hacks (c.f., `raw._orig_units`) to get around the issue of writing to certain formats like `edf`.
         - Promotes writing to a single file format and standardization
    - Cons
         - "two ways of doing the same thing" (same thing being writing BIDS compatible folders in `fif` format)
         -  `fif` format is not supported by BIDS-EEG and BIDS-iEEG
         - more work to implement
         - would limit adoption + user base of `mne-bids` as many people (cc @sappelhoff @monkeyman192) may want to retain original format

all this is to just make sure you've thought through the different pros and cons. For me, it's easiest to implement the kwargs alternative :-)

Reply:
Just to illustrate, one function call from test currently looks like this:

```py
    raw_to_bids(subject_id=subject_id, session_id=session_id, run=run,
               task=task, acquisition=acq, raw_file=raw_fname,
               config=config_fname, hsp=headshape_fname,
               output_path=output_path)
```

This is not going to change even if we use `kwargs` ...